lib/rouge/lexers/sas.rb

```diff
@@ -0,0 +1,568 @@
+          PANEL PARETO PDLREG PDS PDSCOPY PHREG PLAN PLM PLOT PLS PM PMENU
+          POWER PRESENV PRINCOMP PRINQUAL PRINT PRINTTO PROBIT PROTO PRTDEF
+          PRTEXP PSMOOTH PWENCODE QDEVICE QLIM QUANTLIFE QUANTREG QUANTSELECT
+          QUEST RANK RAREEVENTS RDC RDPOOL RDSEC RECOMMEND REG REGISTRY RELEASE
+          RELIABILITY REPORT RISK ROBUSTREG RSREG SCAPROC SCORE SEQDESIGN
+          SEQTEST SERVER SEVERITY SGDESIGN SGPANEL SGPLOT SGRENDER SGSCATTER
+          SHEWHART SIM2D SIMILARITY SIMLIN SIMNORMAL SOAP SORT SOURCE SPECTRA
+          SPP SQL SQOOP SSM STANDARD STATESPACE STDIZE STDRATE STEPDISC STP
+          STREAM SUMMARY SURVEYFREQ SURVEYIMPUTE SURVEYLOGISTIC SURVEYMEANS
+          SURVEYPHREG SURVEYREG SURVEYSELECT SYSLIN TABULATE TAPECOPY TAPELABEL
+          TEMPLATE TIMEDATA TIMEID TIMEPLOT TIMESERIES TPSPLINE TRANSPOSE
+          TRANSREG TRANTAB TREE TSCSREG TTEST UCM UNIVARIATE UPLOAD VARCLUS
+          VARCOMP VARIOGRAM VARMAX VASMP X11 X12 X13 XSL
+        )
+      end
+
+      state :basics do
+        # Rules to be parsed before the keywords (which are different depending
+        # on the context)
+
+        rule /\s+/m, Text
+
+        # Single-line comments (between * and ;) - these can actually go onto multiple lines
+        # case 1 - where it starts a line
+        rule /^\s*%?\*[^;]*;/m, Comment::Single
+        # case 2 - where it follows the previous statement on the line (after a semicolon)
+        rule /(;)(\s*)(%?\*[^;]*;)/m do
+          groups Punctuation, Text, Comment::Single
+        end
+
+        # True multiline comments!
+        rule %r(/[*].*?[*]/)m, Comment::Multiline
+
+        # date/time constants (Language Reference pp91-2)
+        rule /'[0-9a-z]+?'d/i, Literal::Date
+        rule /'.+?'dt/i, Literal::Date
+        rule /'[0-9:]+?([a|p]m)?'t/i, Literal::Date
+
+        rule /'/, Str::Single, :single_string
+        rule /"/, Str::Double, :double_string
+        rule /&[a-z0-9_&.]+/i, Name::Variable
+
+        # numeric constants (Language Reference p91)
+        rule /\d[0-9a-f]*x/i, Num::Hex
+        rule /\d[0-9e\-.]+/i, Num # scientific notation
+
+        # auto variables from DATA step (Language Reference p46, p37)
+        rule /\b(_n_|_error_|_file_|_infile_|_msg_|_iorc_|_cmd_)\b/i, Name::Builtin::Pseudo
+
+        # auto variable list names
+        rule /\b(_character_|_numeric_|_all_)\b/i, Name::Builtin
+
+        # datalines/cards etc
+        rule /\b(datalines|cards)(\s*)(;)/i do
+          groups Keyword, Text, Punctuation
+          push :datalines
+        end
+        rule /\b(datalines4|cards4)(\s*)(;)/i do
+          groups Keyword, Text, Punctuation
+          push :datalines4
+        end
+
+
+        # operators (Language Reference p96)
+        rule %r(\*\*|[\*/\+-]), Operator
+        rule /[^¬~]?=:?|[<>]=?:?/, Operator
+        rule /\b(eq|ne|gt|lt|ge|le|in)\b/i, Operator::Word
+        rule /[&|!¦¬∘~]/, Operator
```

Comment:
Are these symbols all correct?

Reply:
I think so - here is the [SAS documentation about operators](https://documentation.sas.com/?docsetId=lrcon&docsetTarget=p00iah2thp63bmn1lt20esag14lh.htm&docsetVersion=9.4&locale=en) which I was working from.

* line 427: matching ** (exponentiation) or \ * + -
* line 428: matching ^=, ¬=, ~= as variants of 'not equal to' (see the Comparison Operators table in the linked page), as well as >=, >, <=, <, =. And then you can add a colon modifier to any of these to compare only a specified prefix :thinking: 
* line 429: mnemonic equivalents, as shown in the 'Comparison Operators' table in the linked page
* line 430: these are in the Logical Operators table in the linked page. There are lots of NOT alternatives!